generators/app/templates/webapp/Component.ts

```diff
@@ -3,9 +3,11 @@ import models from "./model/models";
 import FlexibleColumnLayoutSemanticHelper from "sap/f/FlexibleColumnLayoutSemanticHelper";
 import { LayoutType } from "sap/f/library";
 import FlexibleColumnLayout from "sap/f/FlexibleColumnLayout";
-import JSONModel from "sap/ui/model/json/JSONModel";
+import JSONModel from "sap/ui/model/json/JSONModel";<% if (gte11170) { %>
```

Comment:
I think you mixed up the version checks: the types for specific events exist since 1.115. It's the generics for getSource which only exist since 1.117.

Reply:
I adapted the version for this. Do I need to do something for the generics of getSource?

Reply:
Well... getSource() always returned the type sap.ui.base.EventProvider before 1.117 and now returns the type on which the specific event was defined. You don't need to do anything to profit from this, but be aware that in older versions you might need more type casting on the result of getSource(), depending on what you do with it.

I mainly mentioned it because in some place you had used the generics yourself to avoid casting the result. Because I wanted to avoid that you accidentally do this in a version before 1.117.